lib/wakes.rb

```diff
@@ -39,8 +39,12 @@ def self.create!(*args)
     build(*args).tap(&:save!)
   end
 
-  def self.redirect(source, target, label = nil)
-    RedirectMapper.new(source, target, label)
+  def self.redirect(source_path, target_path, label = nil)
+    RedirectMapper.new(source_path, target_path, label)
+  end
+
+  def self.redirect_with_hosts(source_host: nil, source_path:, target_host: nil, target_path:, label: nil)
```

Comment:
@benhutton `www.domain.com/source` would be an example. It's not a valid URI, and the Ruby URI method chokes on it, hence "quasi-URI".

I like it for readability in the specs, but wasn't sure I was ready to make it an API.

Anyway, with the code in its present state, we now support all three cases (as well as the reverse directions).